compiler/rustc_attr/src/builtin.rs

```diff
@@ -92,7 +92,11 @@ impl Stability {
 #[derive(HashStable_Generic)]
 pub struct ConstStability {
     pub level: StabilityLevel,
-    pub feature: Symbol,
+    /// This can be `None` for functions that do not have an explicit const feature.
+    /// We still track them for recursive const stability checks.
+    pub feature: Option<Symbol>,
```

Comment:
FWIW I have some ideas for how to get of this `Option` again, but they can only really be implemented together with changes in how we handle `const fn` in `force_unstable_if_unmarked` functions. So I'd prefer to defer that to the next PR in this series.